### PR TITLE
Support different persistence providers in tests

### LINKED ISSUE #29: Enable the possibility to run provider tests
At the moment most of the tests are running hard coded with Hibernate and ignores the selected profile.

This issue should improve the situation by supporting different persistence providers.

The tests have to be migrated step wise in further issues.

### PR DESCRIPTION
**Pull request by CSchulz**
fixes #29

### CODE REVIEW

**Comment by CSchulz** [comment]
@ArneLimburg Finally I have done it ... I have disabled a lot of tests and created a bunch of issues for it.

Please review my changes, I am not happy with all of them f.e. renaming test classes.

File: pom.xml [hunk 1]
```diff
@@ -550,27 +550,21 @@
...
-            <version>1.0.0.Final</version>
+            <groupId>org.eclipse.persistence</groupId>
+            <artifactId>javax.persistence</artifactId>
+            <version>2.1.0</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.hibernate</groupId>
-            <artifactId>hibernate-entitymanager</artifactId>
+            <artifactId>hibernate-core</artifactId>
```
[the remark this answers is not in the document]
Comment (CSchulz) [reply pom.xml]: Finally yes, but there are at the moment some dependencies on this classes f.e. one test unwraps the Hibernate query.

File: src/test/java/org/jpasecurity/persistence/QueryNullEnumTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if (entityManager.getTransaction().isActive()) {
+            entityManager.getTransaction().rollback();
+        }
+        entityManager.close();
+        entityManager.getEntityManagerFactory().close();
+    }
+
+    @Test
+    public void count() {
+        List<DummyEnumClass> result = entityManager.createQuery("SELECT t FROM DummyEnumClass t WHERE t.status = null",
```
[the remark this answers is not in the document]
Comment (ArneLimburg) [reply src/test/java/org/jpasecurity/persistence/QueryNullEnumTest.java]: Can be done in the other Pull-Request

File: pom.xml [hunk 3]
```diff
@@ -550,27 +550,21 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.hibernate.javax.persistence</groupId>
-            <artifactId>hibernate-jpa-2.1-api</artifactId>
-            <version>1.0.0.Final</version>
+            <groupId>org.eclipse.persistence</groupId>
+            <artifactId>javax.persistence</artifactId>
```
[the remark this answers is not in the document]
Comment (CSchulz) [reply pom.xml]: One reason is the vendor specific implementation how to handle persistence providers:
https://github.com/hibernate/hibernate-jpa-api/blob/2.1/src/main/java/javax/persistence/spi/PersistenceProviderResolverHolder.java#L145

Another reason is the different behavior between java 7 and 8.

Furthermore they are stating you should use this dependency at least beginning with 2.2.

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (CSchulz) [reply pom.xml]: I am not sure if we should introduce test cases for all implementations or should rely on one for all.

[the remark this answers is not in the document]
Comment (ArneLimburg) [reply pom.xml]: OK, fine for now with that expanation. We have to work that out.

[the remark this answers is not in the document]
Comment (CSchulz) [reply pom.xml]: Created an issue for it, #66

**Pull request merged**